content/chainguard/chainguard-enforce/troubleshooting-tips.md

```diff
@@ -18,33 +18,193 @@ toc: true
+
+```sh
+chainctl cluster install --group=$GROUP --private --context demo 
 ```
 
-By default, your next command passed without the `-v` flag will revert to the standard `chainctl` experience. Alternatively, you can pass `-v=0` to your `chainctl` command for the default experience, without logging.
+The error will be similar to the below.
 
-## How to disable admission control
+```
+...
+Installing Chainguard agent...
+couldn't fetch cluster OIDC configuration exit status 1
+```
+
+The cause of this may be due to the `openid-configuration` and `jwks` being inaccessible within the Kubernetes API. You can determine this with `kubectl`.
+
+First, check whether the issue is with the `openid-configuration`.
+
+```sh
+kubectl get --raw /.well-known/openid-configuration
+```
+
+If this is the issue, you'll receive an error as output.
+
+```
+Error from server (NotFound): the server could not find the requested resource
+```
+
+Next, you can check if the issue is related to `jwks`.
+
+```sh
+kubectl get --raw /openid/v1/jwks
+```
+
+If `jwks` being inaccessible is the issue, you'll receive an error here.
+
+```
+Error from server (NotFound): the server could not find the requested resource
+```
+
+The output above reveals that the endpoints are not available to `chainctl`, whcih may be due to an auth proxy between `chainctl` and the controlplane.
+
+To solve for this issue, you'll need an OpenID Connect Well-Known Configuration Endpoint, which should be available at the `/.well-known/openid-configuration` path, This will enable you to use `chainctl` for installation and to register an identity. You can check that the file is there with `kubectl`.
+
+```sh
+kubectl get --raw /.well-known/openid-configuration
+```
+
+If you receive an error that the requested resource was not found, make sure you are using the `--private` option on installation. You can review guidance on this option in the [Configuring Enforcer Options](/chainguard/chainguard-enforce/installation/configure-enforcer-options-during-installation/#install-with-chainctl) guide.
+
+If you are using Rancher, you'll be able to generate “direct access” kubeconfigs that bypass the auth proxy and go directly to the cluster. Please follow the [Rancher documentation to create a new kubeconfig file](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/manage-clusters/access-clusters/use-kubectl-and-kubeconfig#authenticating-directly-with-a-downstream-cluster).
+
+### Installing Enforce with Rancher
+
+If you're using Rancher for Kubernetes, you'll need to point directly to your kubeconfig file:
+
+```sh
+KUBECONFIG=/etc/rancher/rke2/rke2.yaml chainctl cluster install --group=$GROUP --private
+```
+
+If you cannot reach the OIDC endpoint, this will cause issues for a cluster install, review the [section above](#cluster-oidc-configuration-exit-status) for guidance on troubleshooting this. 
```

Comment:
```suggestion
If you cannot reach the OIDC endpoint, this will cause issues for a cluster install, review the [previous section](#cluster-oidc-configuration-exit-status) for guidance on troubleshooting this. 
```